docs/serving/samples/hello-world/helloworld-java-spring/README.md

```diff
@@ -112,83 +103,139 @@ cd knative-docs/docs/serving/samples/hello-world/helloworld-java-spring
-folder) you're ready to build and deploy the sample app.
+**NOTE:** Use Docker to build the sample code into a container. To build and push with Docker Hub, run these commands replacing `{username}` with your Docker Hub username.
 
-1. Use Docker to build the sample code into a container. To build and push with
-   Docker Hub, run these commands replacing `{username}` with your Docker Hub
-   username:
+1. Use Docker to build the sample code into a container, then push the container to the Docker registry:
 
-   ```shell
+   ```bash
    # Build the container on your local machine
    docker build -t {username}/helloworld-java-spring .
 
    # Push the container to docker registry
    docker push {username}/helloworld-java-spring
    ```
+## Deploying the app
+
+After the build has completed and the container is pushed to Docker Hub, you can deploy the app into your cluster.
+
+
+=== "yaml"
+
+       1. Create a new file, `service.yaml` and copy the following service definition
+          into the file. Make sure to replace `{username}` with your Docker Hub
+          username.
+
+          ```yaml
+          apiVersion: serving.knative.dev/v1
+          kind: Service
+          metadata:
+            name: helloworld-java-spring
+            namespace: default
+          spec:
+            template:
+              spec:
+                containers:
+                  - image: docker.io/{username}/helloworld-java-spring
+                    env:
+                      - name: TARGET
+                        value: "Spring Boot Sample v1"
+          ```
+
+       Ensure that the container image value
+       in `service.yaml` matches the container you built in the previous step. Apply
+       the configuration using `kubectl`:
+
+       ```bash
+       kubectl apply --filename service.yaml
+       ```
+
+
+=== "kn"
+
+       With `kn` you can deploy the service with
+
+       ```bash
+       kn service create helloworld-java-spring --image=docker.io/{username}/helloworld-java-spring --env TARGET="Spring Boot Sample v1"
+       ```
+
+       This will wait until your service is deployed and ready, and ultimately it will print the URL through which you can access the service.
+
+
 
-1. After the build has completed and the container is pushed to docker hub, you
-   can deploy the app into your cluster. Ensure that the container image value
-   in `service.yaml` matches the container you built in the previous step. Apply
-   the configuration using `kubectl`:
 
-   ```shell
-   kubectl apply --filename service.yaml
-   ```
 
-1. Now that your service is created, Knative will perform the following steps:
+   During the creation of your service, Knative performs the following steps:
 
    - Create a new immutable revision for this version of the app.
-   - Network programming to create a route, ingress, service, and load balancer
+   - Network programming to create a route, ingress, service, and load balance
```

Comment:
```suggestion
   - Network programming to create a Route, ingress, Service, and load balancer.
```

Not sure about this, it looks weird when we give caps to some resources, e.g. Services and Routes, but not others.

Reply:
huh... yeah that is a bit weird...

Reply:
to be fair, ingress and load balancer are a type of thing, Route and Service are proper Knative nouns